advanced-solidus/images-and-image-processing.md

```diff
@@ -0,0 +1,196 @@
+# Images and Image Processing
+
+Solidus utilizes [MiniMagick](https://github.com/minimagick/minimagick) for [ActiveStorage](https://edgeguides.rubyonrails.org/active_storage_overview.html) and [ImageMagick](https://imagemagick.org/index.php) for [Paperclip](https://github.com/thoughtbot/paperclip#paperclip) to handle the under-the-hood resizing and image processing. After an image is uploaded to a project, it is automatically resized based on when the view requests an image variant "style". The main dictating functions can be found in [adapter](https://github.com/solidusio/solidus/blob/master/core/app/models/concerns/spree/active_storage_adapter/attachment.rb) that controls sizing and the [ActiveStorage](https://github.com/solidusio/solidus/blob/master/core/app/models/spree/image/active_storage_attachment.rb)/[Paperclip](https://github.com/solidusio/solidus/blob/master/core/app/models/spree/image/paperclip_attachment.rb) attachment modules. Solidus is currently in the process of depreciating Paperclip due to ActiveStorage's inclusion with rails.
+
+## Customizing Size Defaults and Attachment Module Override
+
+As you can see from the linked modules above, the default configuration provides only a few variant sizes out of the box. Changing the image sizing is simply a configuration of an attachment module and is referenced when the attachment class method `style_to_size` is called. We could create ourselves a new adapter for our app and override the settings that way, but in this case, we just want to add some new types of sizings for our UI display.
```

Comment:
We linked a lot of stuff, and the variants will not be visible in those if/when https://github.com/solidusio/solidus/pull/4062 will be merged, isn't it? I guess we can just say something like:


> By default Solidus comes with these image format: .... that are accessible as `image.url(:mini)` where `image` is an instance of `Spree::Image` and `:mini` is the format that you need. 

> You can easily configure the images' sizes that will be available for your store in order to match your design and frontend needs.

Reply:
I like the suggestion and added roughly the same info, WDYT?

By default Solidus comes with these image formats:... (formats in code block)

These formats are accessible by calling image.url(:mini) in the view where image is an instance of Spree::Image and :mini is the "style" format that you need.